### DISCUSSION TITLE
Not working with STMF4

### DISCUSSION THREAD

**Discussion by jadczakoff** [hathach/tinyusb#1682] [Q&A] [answered]
I'm trying to add tinyUSB library but I get this define CFG_TUSB_RHPORT1_MODE as not defined. If i comment this line, my usb is not appear in device manager. I did this tutorial: https://anvilelectronics.ovh/tinyusb-na-stm32-device-cdc/. Could you check what i'm doing wrong ?
At the end I want to make CDC communication without misc.
![image](https://user-images.githubusercontent.com/71470597/195599914-0259f395-2968-4fdf-91d3-7f5c4f4ada5c.png)

Link to repo: https://github.com/jadczakoff/Test_tinyUSB_v3

**Comment by jadczakoff** [comment, marked as answer]
Okey I solved issue. Silly I did't think to change number from 1 to 0... so it looks now:
  if (!(rhport == 1 && (CFG_TUSB_RHPORT0_MODE & OPT_MODE_HIGH_SPEED))) usb_otg->GCCFG |= USB_OTG_GCCFG_PWRDWN;

and in tusb_config.h I added:
#define CFG_TUSB_MCU                 OPT_MCU_STM32F4
#define CFG_TUSB_OS                  OPT_OS_NONE
#define BOARD_DEVICE_RHPORT_SPEED    OPT_MODE_FULL_SPEED
#define BOARD_DEVICE_RHPORT_NUM      0
#define CFG_TUSB_RHPORT0_MODE         (OPT_MODE_DEVICE | OPT_MODE_FULL_SPEED)
Now its working :D Discussion can be closed

**Reply by hathach** [reply]
thanks for update, I have marked this as anwser
